### PR TITLE
Add Radial fill to yard scouting

### PR DESCRIPTION
**Pull request by bpage-gamedev**
I added a radial fill image to the Dog Image prefab, and added a few
lines of code to dogcollarslot.cs. It piggybacks on the existings handle
and init methods. I tagged my work with "BP" out of habit. If this is
too much commenting, let me know. The coroutine lerps the image down
each second for a second.

### CODE REVIEW

**Comment by imann24** [comment]
@bpage-gamedev This looks very good, however, there is a notable bug:

The visual timer resets if I switch scenes (click the door to switch between yard and living room. It should always be in sync with the time remaining divided by total time

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs [hunk 1]
```diff
@@ -251,7 +267,30 @@ void handleDogTimerChange(Dog dog, float timeRemaining)
         if (timerText && !dog.HasRedeemableGift)
         {
             timerText.text = dog.RemainingTimeScoutingStr;
+
+            //BP Instead of doing the below line, start a lerp every second that lerps the radial fill down a second
+            //radialFill.fillAmount = timeRemaining / dog.Info.TotalTimeToReturn;
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Remove commented out code

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs [hunk 2]
```diff
@@ -225,6 +236,11 @@ void handleDogGiftEvents(string eventName, CurrencyData gift)
 
     void handleGiftFound(CurrencyData gift)
     {
+        // BP gift has been found, so deactivate the radial fill image
+        if (radialFill != null)
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Remove space between `if` and parens

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs [hunk 3]
```diff
@@ -251,7 +267,30 @@ void handleDogTimerChange(Dog dog, float timeRemaining)
...
+
         }
     }
+    
+    //BP Makes transition smooth from second to second
+    IEnumerator lerpRadial(float timeRemaining, float totalTime)
+    {
+        float startPoint = timeRemaining / totalTime;
+        float endPoint = (timeRemaining - 1) / totalTime;
+        float t = 0;
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Use a more expressive var name

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs [hunk 4]
```diff
@@ -251,7 +267,30 @@ void handleDogTimerChange(Dog dog, float timeRemaining)
...
+        float currentPoint = startPoint;
 
+        while (t < 1)
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: This should not be hardcoded

Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Remove space between `while` and paren

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs [hunk 5]
```diff
@@ -251,7 +267,30 @@ void handleDogTimerChange(Dog dog, float timeRemaining)
         if (timerText && !dog.HasRedeemableGift)
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Remove space between `if` and paren

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs [hunk 6]
```diff
@@ -108,7 +114,7 @@ protected override void handleChangeDog(Dog previousDog)
     protected override void callOnOccupiedSlotClick(Dog dog)
     {
         // Safeguard against opening up tons of copies of the panel
-        if(!redeemDisplayIsOpen)
+        if (!redeemDisplayIsOpen)
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Remove space between `if` and paren

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs [hunk 7]
```diff
@@ -1,10 +1,13 @@
 ﻿/*
- * Author: Grace Barrett-Snyder 
+ * Authors: Grace Barrett-Snyder, Ben Page
  * Description: Controls a DogSlot for a Dog that's outside (has name and timer).
  */
 
 using UnityEngine;
 using UnityEngine.UI;
+using System;
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: What are you using this library for?

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs [hunk 8]
```diff
@@ -1,10 +1,13 @@
...
+using System.Collections;
+using System.Collections.Generic;
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: What are you using this library for?

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs (hunk 7 above)
Comment (bpage-gamedev) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: I wasn't, taken care of.

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs (hunk 8 above)
Comment (bpage-gamedev) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: System.Collections for the IEnumerator I added.
Wasn't using '.Generic', taken care of.

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs (hunk 6 above)
Comment (bpage-gamedev) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Done.

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs (hunk 2 above)
Comment (bpage-gamedev) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Done.

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs (hunk 5 above)
Comment (bpage-gamedev) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Done.

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs (hunk 1 above)
Comment (bpage-gamedev) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Done.

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs (hunk 3 above)
Comment (bpage-gamedev) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Changed 't' to 'lerpTime'.

File: Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs (hunk 4 above)
Comment (bpage-gamedev) [inline Assets/Scripts/UI/PickupPup/DogSlot/DogCollarSlot.cs]: Got rid of 'currentPoint' altogether and removed space

**Pull request merged**